### ISSUE TITLE
[Feature Request]: Should warn about missing `await` on promise 

### ISSUE THREAD

**Issue by nurmohammed840** [denoland/deno_lint#1121]
Forgetting `await` keyword on promise is common source of bug. And deno lint should warn about it.

And deno lint shouldn't annoy people.

```typescript
interface Write {
    write(bytes: Uint8Array): Promise<number>
}
class Foo implements Write {
    async write(_: Uint8Array) {   // Warn: Async method 'write' has no 'await' expression
        return 0
    }
}
```

Here `async` keyword is required by interface `Write`

**Comment by magurotuna** [comment]
We would love to have that feature, but unfortunately that requires type information which deno_lint cannot utilize now.

I'm going to close this issue as it's duplicate of https://github.com/denoland/deno_lint/issues/303
